### PR TITLE
Adds a new power storage type: The Megacell. Drastically reduces power cell consumption/storage. [MDB Ignore]

### PR DESCRIPTION
**Pull request by Watermelon914**
## About The Pull Request
As the title says. A standard power cell now only stores 10 KJ and drains power similar to how it did before the refactor to all power appliances.

The new standard megacell stock part stores 1 MJ (what cells store right now). APCs and SMESs have had their power cells replaced with these megacell stock parts instead. Megacells can only be used in APCs and SMESs. It shouldn't be possible to use megacells in any typical appliance.

This shouldn't change anything about how much 'use' you can get out of a power cell in regular practice. Most should operate the same and you should still get the same amount of shots out of a laser gun, and we can look at expanding what can be switched over to megacells, e.g. if we want mechs to require significantly more power than a typical appliance.

Thanks to Meyhazah for the megacell icon sprites.

## Why It's Good For The Game
Power cell consumption is way too high ever since the power appliance refactor that converted most things to be in joules. It's a bit ridiculous for most of our machinery to drain the station's power supply this early on. 

The reason it's like this is because regular appliances (laser guns, borgs, lights) all have a cell type that is identical to the APC/SMES cell type. And it means that if we want to provide an easy way to charge these appliances without making it easy to charge APCs/SMESs through a power bug exploit, we need to introduce a new cell type to differentiate between what supplies power and regular appliances that use power. This is primarily what the megacell stock part does.

This moves us back to what it was originally like before the power refactor, where recharging power cells wouldn't drain an exorbitant amount of energy. However, it maintains the goal of the original refactor which was to prevent people from cheesing power generation to produce an infinite amount of power, as the power that APCs and SMESs operate at is drastically different from the power that a regular appliance uses.

## Changelog
:cl: Watermelon, Mayhazah
balance: Drastically reduces the power consumption and max charge of power cells
balance: Added a new stock part called the battery, used primarily in the construction of APCs and SMESs.
add: Suiciding with a cell/battery will shock you and potentially dust you/shock the people around you if the charge is great enough.
/:cl:

### CODE REVIEW

**Comment by carlarctg** [comment]
very silly name. everyone will be confused by it

**Comment by CRITAWAKETS** [comment]
Honestly SMES units probably should just be changed to not require any cells at all, considering they're only going to get filled with the Super Special Battery Cells anyway. SMES in real-life doesn't use batteries so, y'know, realism to justify a gameplay improvement.

**Comment by excalisword** [comment]
Will cyborgs also use these new batteries?

**Comment by Jacquerel** [comment]
"Power Cells" sound bigger and cooler than batteries so I almost wonder if the names should be exchanged.

**Comment by Watermelon914** [comment]
> Will cyborgs also use these new batteries?

No

**Comment by Watermelon914** [comment]
> Honestly SMES units probably should just be changed to not require any cells at all, considering they're only going to get filled with the Super Special Battery Cells anyway. A laser part instead could determine the max charge. SMES in real-life doesn't use batteries so, y'know, realism to justify a gameplay improvement.

It's not just SMESs, APCs also need a special cell for this reason. It's easier to use cells because they provide an easy interface that allows you to store power

**Comment by Deadgebert** [comment]
Would it be possible to still allow regular power cells to work in the APCs? The 'station is going to shit and i need emergency power' is pretty fun sandbox solution.

**Comment by Watermelon914** [comment]
> Would it be possible to still allow regular power cells to work in the APCs? Slotting in a power cell during those 'station is going to shit and i need emergency power' moments is a pretty fun/cool sandbox solution.

sure

**Comment by LemonInTheDark** [comment]
I have BEEN saying this

**Comment by optimumtact** [comment]
Can you implement hte rest of my power delivery overhaul

**Comment by Watermelon914** [comment]
> Can you implement hte rest of my power delivery overhaul

where's the doc? I can maybe take a look

**Comment by Wallemations** [comment]
My concern is with these still being "cells" rather than batteries, which might confuse some players as to what they are used for.

Don't we already have a big car battery in the game for general maint junk? Maybe we could give that some use?

**Comment by Xkallubar** [comment]
I have a feeling this is going to screw up Technophile chaplain sect favor gain.

**Comment by Watermelon914** [comment]
> My concern is with these still being "cells" rather than batteries, which might confuse some players as to what they are used in.
> 
> Don't we already have a big car battery in the game for general maint junk? Maybe we could give that some use? I find the concept that these industrial-sized batteries are more primitive than the sleeker, publicly-used cells interesting personally.

They do look noticeably different in game and are called megacells. I don't think it'll be that confusing.
image

**Comment by Watermelon914** [comment]
> I have a feeling this is somehow going to screw up Technophile chaplain sect favor gain.

If you think it's going to do that, then test it yourself and post it here. It's unlikely though

**Comment by ATH1909** [comment]
How does this interact with ethereals and their ability to recharge/drain APCs?

And with inducers, for that matter?

**Comment by Pickle-Coding** [comment]
This will be weird with SMES. The roundstart ones will start with 50MJ because they're mapped in. Building new ones with the same parts will cause them to only be able to store 5MJ with them, which would last only 25 seconds with 200kW output. This will also have a massive impact on roundstart cell drain. Current APCs have cells with 2.5MJ in them. If they got downgraded to 1MJ cells, then they'd only have 40% of the time to spend before running out of energy. We should be careful with this, because we don't have much room before starting power becomes a rush for engineers.

I suggest making megacells have a capacity of 2.5MJ to keep APCs consistent, then make SMES require 20 cells so they become consistent too. Not as important, but I think the new standard cell capacities are a bit low. They don't have to be powersinks, but I do want them to be noticeable on the APC load when someone decides to charge something. 10kJ would just be a binary matter of whether we have power or we don't have power, rather than having enough power to deal with the load changes.

File: code/modules/projectiles/guns/energy.dm [hunk 1]
```diff
@@ -5,8 +5,8 @@
 	icon = 'icons/obj/weapons/guns/energy.dmi'
 
 	/// What type of power cell this uses
-	var/obj/item/stock_parts/cell/cell
-	var/cell_type = /obj/item/stock_parts/cell
+	var/obj/item/stock_parts/power_store/cell
```
Comment (SyncIt21) [inline code/modules/projectiles/guns/energy.dm:8]:
```suggestion
	var/obj/item/stock_parts/power_store/cell/cell
```
We don't want mega cells in here

File: code/modules/reagents/chemistry/machinery/chem_dispenser.dm [hunk 2]
```diff
@@ -11,13 +11,13 @@
 	processing_flags = NONE
 
 	/// The cell used to dispense reagents
-	var/obj/item/stock_parts/cell/cell
-	/// Efficiency used when converting cell power to reagents. Units (volume) per joule.
-	var/powerefficiency = 1e-4
+	var/obj/item/stock_parts/power_store/cell
```
Comment (SyncIt21) [inline code/modules/reagents/chemistry/machinery/chem_dispenser.dm:14]:
```suggestion
	var/obj/item/stock_parts/power_store/cell/cell
```

File: code/modules/reagents/chemistry/machinery/chem_dispenser.dm [hunk 3]
```diff
@@ -422,20 +422,20 @@
...
 	visible_message(span_danger("[src] malfunctions, spraying chemicals everywhere!"))
 
 /obj/machinery/chem_dispenser/RefreshParts()
 	. = ..()
 	recharge_amount = initial(recharge_amount)
-	var/newpowereff = INVERSE(1.5e4)
+	var/new_power_cost = initial(power_cost)
 	var/parts_rating = 0
-	for(var/obj/item/stock_parts/cell/stock_cell in component_parts)
+	for(var/obj/item/stock_parts/power_store/stock_cell in component_parts)
```
Comment (SyncIt21) [inline code/modules/reagents/chemistry/machinery/chem_dispenser.dm:435]:
```suggestion
	for(var/obj/item/stock_parts/power_store/cell/stock_cell in component_parts)
```

File: code/modules/reagents/reagent_containers/misc.dm [hunk 4]
```diff
@@ -8,13 +8,13 @@
 	reagent_flags = OPENCONTAINER
 	fill_icon_state = "maunafilling"
 	fill_icon_thresholds = list(25)
-	var/obj/item/stock_parts/cell/cell
+	var/obj/item/stock_parts/power_store/cell
```
Comment (SyncIt21) [inline code/modules/reagents/reagent_containers/misc.dm:11]:
```suggestion
	var/obj/item/stock_parts/power_store/cell/cell
```

File: code/modules/religion/religion_sects.dm [hunk 5]
```diff
@@ -146,7 +146,7 @@
 	do not heal organic limbs. You can now sacrifice cells, with favor depending on their charge."
 	tgui_icon = "robot"
 	alignment = ALIGNMENT_NEUT
-	desired_items = list(/obj/item/stock_parts/cell = "with battery charge")
+	desired_items = list(/obj/item/stock_parts/power_store = "with battery charge")
```
Comment (SyncIt21) [inline code/modules/religion/religion_sects.dm:149]:
```suggestion
	desired_items = list(/obj/item/stock_parts/power_store/cell = "with battery charge")
```

**Review by SyncIt21 — COMMENTED** [review]
Some cell type path corrections. We don't want batteries to be inserted here

**Comment by GPeckman** [comment]
> I have a feeling this is somehow going to screw up Technophile chaplain sect favor gain.

On the contrary, this is going to fix it. As it currently stands, chaplains need about 300 megajoules for a single conversion ritual. That takes exactly as long as you'd expect it to take with a basic SM setup. Afterwards, it'll only take a much more modest 3 megajoules.

File: code/modules/power/power.dm [hunk 6]
```diff
@@ -251,7 +251,7 @@
  * - channel: The power channel to use.
  * Returns: The amount of energy the cell received.
  */
-/obj/machinery/proc/charge_cell(amount, obj/item/stock_parts/cell/cell, grid_only = FALSE, channel = AREA_USAGE_EQUIP)
+/obj/machinery/proc/charge_cell(amount, obj/item/stock_parts/power_store/cell/cell, grid_only = FALSE, channel = AREA_USAGE_EQUIP)
```
Comment (Pickle-Coding) [inline code/modules/power/power.dm:254]:
```suggestion
/obj/machinery/proc/charge_cell(amount, obj/item/stock_parts/power_store/cell, grid_only = FALSE, channel = AREA_USAGE_EQUIP)
```

**Comment by Loafin34** [comment]
how would we charge megacells? from what i could glean from the testmerge, cell chargers don't work with it

**Comment by SyncIt21** [comment]
> how would we charge megacells? from what i could glean from the testmerge, cell chargers don't work with it

These are not meant to be charged manually outside of SMES and APC.

**Comment by carlarctg** [comment]
can i throw these into the ocean

**Comment by alien3301** [comment]
Inducers should be able to take megacells and the roundstart engi inducers should start with the same megacells APCs start with. You should also be able to manually charge a megacell by using an inducer on it (this might already be the case but I have a feeling you can't charge cells outside machinery like that).

**Comment by alien3301** [comment]
Also mechs should definetely use megacells cause they're huge and using the kind of battery you use in your phone (PDA) for them is kinda silly.

**Review by Maurukas — APPROVED** [review]
Thank you for the Update Paths. Looks like a simple replacement in the changes.

**Comment by Jacquerel** [comment]
> Also mechs should definetely use megacells cause they're huge and using the kind of battery you use in your phone (PDA) for them is kinda silly. And if mechs will use those the robo fabricators should probably be able to print them.

Nothing which isn't an APC can be allowed to use megacells or we are reintroducing the problems that megacells solve.

**Comment by Watermelon914** [comment]
https://github.com/tgstation/tgstation/assets/37270891/2e8da424-9083-42ca-871f-0c0fee579c64

Added a new suicide interaction with cells

**Pull request merged**